Applications/Spire/Source/Ui/DropDownList.cpp

```diff
@@ -0,0 +1,78 @@
+#include "Spire/Ui/DropDownList.hpp"
+#include <QEvent>
+#include <QHBoxLayout>
+#include "Spire/Ui/ListItem.hpp"
+#include "Spire/Ui/ListView.hpp"
+#include "Spire/Ui/OverlayPanel.hpp"
+#include "Spire/Ui/ScrollableListBox.hpp"
+
+using namespace Spire;
+using namespace Spire::Styles;
+
+namespace {
+  auto SCROLLABLE_LIST_STYLE(StyleSheet style) {
+    style.get(Any()).
+      set(BackgroundColor(QColor::fromRgb(0, 0, 0, 0))).
+      set(border_size(0));
+    return style;
+  }
+
+  auto get_border_size(QWidget& widget) {
+    auto border_size = QSize(0, 0);
+    auto& stylist = find_stylist(widget);
+    auto block = stylist.get_computed_block();
+    for(auto& property : block) {
+      property.visit(
+        [&] (const BorderTopSize& size) {
+          stylist.evaluate(size, [&] (auto size) {
+            border_size.rheight() += size;
+          });
+        },
+        [&] (const BorderRightSize& size) {
+          stylist.evaluate(size, [&] (auto size) {
+            border_size.rwidth() += size;
+          });
+        },
+        [&] (const BorderBottomSize& size) {
+          stylist.evaluate(size, [&] (auto size) {
+            border_size.rheight() += size;
```

Comment:
You can not use capture by reference in a closure whose lifetime exceeds it lexical scope, this is a dangling reference.